worker/worker.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+var (
+	// workerID is the id assigned by the scheduler
+	// after registering on scheduler.
+	workerID string
+)
+
+func init() {
+	loadConfig()
+}
+
+// Entry point of the worker application.
+func main() {
+	go registerWorker()
+
+	sig := make(chan os.Signal)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+
+	for {
+		select {
+		case s := <-sig:
+			fatal(fmt.Sprintf("Signal (%d) received, stopping\n", s))
```

Comment:
I'm not sure I would consider this fatal, this to me looks more like a normal exit, and as such I suspect the program shouldn't exit with code 1.

Reply:
I do agree. This is the only nonfatal exit and the exit code should be different. Changing it now